### DISCUSSION TITLE
previous and next record inside form using actions? or custom fields with button?

### DISCUSSION THREAD

**Discussion by jonbalagot** [filamentphp/filament#7716] [General]
Any pointers is much appreciated.

![Screenshot_20230813_003535](https://github.com/filamentphp/filament/assets/7664287/f8080ed3-c103-4e3a-87c2-e7effe15b341)
![Screenshot_20230813_005239](https://github.com/filamentphp/filament/assets/7664287/7cb1cbbf-4540-4173-a064-8e7c7b56d53b)

**Comment by jonbalagot** [comment]
Created a custom field but returns error "Unable to call component method. Public method not found on component.


UserResource.php

`Forms\Components\TextInput::make('name')`
                   ` ->required()`
                    `->maxLength(255),`
` Forms\Components\TextInput::make('email')`
                  `  ->email()`
                    `->required()`
                   ` ->maxLength(255),`
`Forms\Components\DateTimePicker::make('email_verified_at'),`
`Forms\Components\TextInput::make('password')`
                  `  ->password()`
                   ` ->required()`
                  ` ->maxLength(255),`
 ` Forms\Components\ViewField::make('rating')`
                    `->view('forms.components.next-record-button')`


Forms/Components/NextRecordButton.php

class NextRecordButton extends Field
{
    protected string $view = 'forms.components.next-record-button';

    public function getNextRecord()
    {
        dd('works');
    }

    protected function setUp(): void
    {
        parent::setUp();
    
        $this->registerListeners([
            'getNextRecord' => [
                function (Component $component): void {
                    dd($component);
                    // $component->getActions([
                    //     Action::make('test')->icon('heroicon-s-cog'),
                    // ]);
                },
            ],
        ]);
    }
}


Resources/views/forms/components/next-record-button.blade.php

`<div x-data="{ state: $wire.entangle('{{ $getStatePath() }}') }">
    <x-filament::button x-model="state" wire:click="getNextRecord">
        New user
    </x-filament::button>
</div>`

**Comment by jonbalagot** [comment]
Updated the blade file and error is gone

`<div x-data="{ state: $wire.entangle('{{ $getStatePath() }}').defer }">
    <x-filament::button wire:click="dispatchFormEvent('getNextRecord')" wire:model.defer="{{ $getStatePath() }}">
        New user
    </x-filament::button>
</div>`

**Comment by danharrin** [comment]
You should use actions, not a custom field. Check the Forms>Actions docs

**Reply by jonbalagot** [reply]
thank you sir! I feel like there's too much code on what i've done.  But well, I think it serves the function that I want. 
![Screenshot_20230813_232606](https://github.com/filamentphp/filament/assets/7664287/de05c786-b074-4d99-9fa8-b6ba4ade22be)

**Reply by leandrocfe** [reply]
Are you trying to do something like this?
```php
protected function getHeaderActions(): array
{
    return [

        Action::make('previous')
            ->action(function () {
                $this->record = Model::find($this->record->id - 1);
                $this->refreshFormData(array_keys($this->data));
            })
            ->disabled(fn () => $this->record->id <= Model::oldest()->first()->id),

        Action::make('next')
            ->action(function () {
                $this->record = Model::find($this->record->id + 1);
                $this->refreshFormData(array_keys($this->data));
            })
            ->disabled(fn () => $this->record->id >= Model::latest()->first()->id),
    ];
}
```
